server/server.js

```diff
@@ -0,0 +1,92 @@
+const express = require('express'); // simple http framework
+const bodyParser = require('body-parser'); // middleware help parsing requests
+const morgan = require('morgan'); // http logger
+const uuid = require('uuid'); // unique id generators
+const app = express();
+const MongoClient = require('mongodb').MongoClient
+//const data = {}; // this replace our database for educationnal purposes
+var database;
+
+//===========================================================================
+//database connection Test
+//mongodb://localhost:27017/db
+MongoClient.connect('mongodb://abhawsar:password@ds139322.mlab.com:39322/ayushbhawsar_test_db', function (err, db) {
+    database = db;
+    if (err) throw err
+
+    db.collection('times').find().toArray(function (err, result) {
+        if (err) throw err
+
+        console.log(result)
+    })
+})
+//===========================================================================
+
+app.set('port', (process.env.PORT || 3000));
+app.use(bodyParser.json());
+app.use(morgan('dev'));
+
+app.use(express.static(__dirname + '/public'));
+
+app.get('/api/times', (req, res) => {
+    database.collection('times').find().toArray(function (err, result) {
+        if (err) throw err
+        //console.log(result);
+        res.json(result);
+    })
+});
+
+app.get('/api/times/:id', (req, res) => {
+    database.collection('times').find({ id: req.params.id }).toArray(function (err, result) {
+        if (err) throw err
+        //console.log(result);
+        res.json(result);
+    })
+});
+
+app.post('/api/times', (req, res) => {
+    database.collection('times').insertOne({
+        id: uuid.v1(),
+        time: new Date()
+    })
+        .then(function (result) {
+            res.json(result);
+        })
+});
+
+app.put('/api/times/:id', (req, res) => {
+    //const id = req.params.id;
+    database.collection('times').updateOne(
+        { id: req.param.id },
+        {
+            $set: { time: new Date() },
+            $currentDate: { lastModified: true }
+        }).then(function (result) {
+            res.json(result);
+        }).catch(function (err) {
+            console.log(err)
+        });
+});
+
+app.patch('/api/times/:id', (req, res) => {
+    const id = req.params.id;
+    if (!data[id]) throw new Error(`cannot find data with id ${id}!`);
+    data[id] = Object.assign(data[id], req.body);
+    res.json(data[id]);
+});
```

Comment:
for `PUT` and `PATCH`, you can have the same service.
the difference it's what is being expected to be sent from the client.

for example, in your put, you could have a validation that all required fields are populated, otherwise returns an exception. While patch would be more permissive if not all keys are specified